### ISSUE TITLE
[discussion] Should we use team-discussions instead of maintainers-discussion private repo?

### ISSUE THREAD

**Issue by itaditya** [probot/background-check#46]
As mentioned by @gr2m 
>gregor [11:56 PM]
@dev_aditya you probably discussed that already, but I wonder if there is an alternative to a private repository? I fear it will prevent a lot of open source projects that use a GitHub Organization to use it due to the cost of an upgrade for private repositories. I wonder if the app could post a message to a team chat for example :thinking_face: Not sure if an app can do that

These are all valid points. Would love to know everybody's opinions on this.

/cc @bkeepers @JasonEtco

**Comment by bkeepers** [comment]
This would be a great use of team discussions. If the APIs are supported (I think they are), then it would make sense to avoid the private repo issue that will affect most organizations.

**Comment by itaditya** [comment]
The only downside is that team discussions are not available for individual owner accounts. In that case we will have to fallback to private repo

**Comment by itaditya** [comment]
@gr2m managing an organisation via bot sounds interesting. But I feel doing private repos in it and then adding maintainers will get a little complicated. Is there a way we can utilize team discussions under the app's organisation. This way we can get around the limitation of individual accounts not having team discussion

**Comment by gr2m** [comment]
> Is there a way we can utilize team discussions under the app's organisation. This way we can get around the limitation of individual accounts not having team discussion

Good idea, I think it could work. You would need one team per installation / repository, you would still need to invite people to the background-check organization. But I think it could be done. Once they are invited to the org, the app could use a bot account with admin rights to add them to teams for the different repositories

**Comment by JasonEtco** [comment]
I like this idea. You may also want to leverage the `installation` event to create those teams when the app is installed, to make a better experience for the maintainers.

**Comment by itaditya** [comment]
@gr2m was checking https://developer.github.com/v3/orgs/members/#create-organization-invitation found out that while creating the invitation we can also specify which teams to add the person too

**Comment by itaditya** [comment]
@gr2m in the original implementation, there was one maintainers-discussion repo for the whole org, and so in case the bot finds a hostile user while watching comments on all org repos it opens issue in the one maintainers-discussion repo. In the same I was thinking, instead of one team per repo we should do one team per account (org or individual)

**Comment by gr2m** [comment]
From my experience, the same people who are owners of an organization will be interested in getting the messages by background-check. For repositories that are on my own account I would prefer a team for all collaborators on that repository.

So, in summary, I would suggest

- One team per organization
- One team per repository if installed on an individual account

**Comment by itaditya** [comment]
@gr2m I agree with you

**Comment by itaditya** [comment]
@gr2m I agree with you

**Comment by itaditya** [comment]
@gr2m I agree with you

**Comment by itaditya** [comment]
### One issue with team discussions is that I can no longer check if analysis for a user has already happended.

In the current implementation, I search for a issue titled like itaditya-discussion in maintainers-discussion repo using the github search api.

The search api is not available for team discussions and so I can't search for a thread titled itaditya-discussion. Another implementation for this would be to get a list of all the threads and then check its title but that I think will be a long process and so I don't think we should do that.

Should we stick with having a **probot-background-check** org with paid plan containing repos like **owner-maintainers-discussion** and in that keeping issues like **user-discussion**

**Comment by JasonEtco** [comment]
Hm that's a good point. I definitely think that team discussions is a better tool for this than a private repo, but since they cannot be searched via the API that makes it difficult to work with.

> get a list of all the threads and then check its title

You're right here - you could [list all the discussions](https://developer.github.com/v3/teams/discussions/#list-discussions) (and use `context.github.paginate`), then call `Array.find()` to find the specific one. While I wouldn't normally suggest loading every discussion like this, I think that its a good tradeoff to make use of team discussions.

Ultimately, its a decision between a better experience for the users (using team discussions) or a better experience for you the developer (using a private repo with issues). Phrased that way hopefully you can see which way I'm leaning 😄

**Comment by itaditya** [comment]
@jasonetco I can see in what direction you are leaning. Let me think more about how we can solve the searching problem

**Comment by itaditya** [comment]
@jasonetco @bkeepers I can mix both my implementations to get the best of both worlds. In my probot-background-check I can have a repo per owner and in that repo I'll open a issue for hostile user just like now and then I can search for existing user. But the use of issue ends here. The discussion will be done via team discussions only

**Comment by itaditya** [comment]
So I have worked up the implementation. Check the sample here https://github.com/probot-background-check/probot-test-org-discussions/issues/1

**Comment by itaditya** [comment]
@bkeepers @JasonEtco @gr2m this feature has been implemented.
